dll/shellext/fontext/CFontExt.cpp

```diff
@@ -363,3 +365,176 @@ STDMETHODIMP CFontExt::GetClassID(CLSID *lpClassId)
     return S_OK;
 }
 
+// *** IDropTarget methods ***
+STDMETHODIMP CFontExt::DragEnter(IDataObject* pDataObj, DWORD grfKeyState, POINTL pt, DWORD* pdwEffect)
+{
+    FORMATETC formatETC;
+    HRESULT hr = DoGetFormatETC(pDataObj, formatETC);
+    if (hr != S_OK)
+    {
+        *pdwEffect = DROPEFFECT_NONE;
+        return DRAGDROP_S_CANCEL;
+    }
+
+    // TODO: Check filename extension
+
+#if 1   // Please implement DoGetFontTitle
+    *pdwEffect = DROPEFFECT_NONE;
+    return DRAGDROP_S_CANCEL;
+#else
+    *pdwEffect = DROPEFFECT_COPY;
+    return S_OK;
+#endif
+}
+
+STDMETHODIMP CFontExt::DragOver(DWORD grfKeyState, POINTL pt, DWORD* pdwEffect)
+{
+    return S_OK;
+}
+
+STDMETHODIMP CFontExt::DragLeave()
+{
+    return S_OK;
+}
+
+STDMETHODIMP CFontExt::Drop(IDataObject* pDataObj, DWORD grfKeyState, POINTL pt, DWORD* pdwEffect)
+{
+    FORMATETC formatETC;
+    HRESULT hr = DoGetFormatETC(pDataObj, formatETC);
+    if (hr != S_OK)
+    {
+        *pdwEffect = DROPEFFECT_NONE;
```

Comment:
Same here, should be checked first maybe?

Reply:
OK.